doc/cephfs/kernel.rst

```diff
@@ -1,58 +1,122 @@
 
-Omitting the monitor addresses will cue ``mount.ceph`` to look them up
-in the local configuration file.  If you know at least one of the monitor
-addresses and ports you can specify them directly in a comma-separated
-list, and the kernel will avoid looking them up::
+    # on client host
+    mkdir /etc/ceph
+    ssh {user}@{mon-host} "sudo ceph config generate-minimal-conf" | sudo tee /etc/ceph/ceph.conf
 
-        # mount -t ceph 192.168.0.1:6789,192.168.0.2:6789:/ /mnt/mycephfs
+   Alternatively, you may copy the conf file. But the above method creates a
+   conf with minimum details which is better.
 
-To mount a subtree of the cephfs root, append the path to the device
-string ::
+#. Ensure that the conf file has appropriate permissions::
 
-        # mount -t ceph :/subvolume/dir1/dir2  /mnt/mycephfs -o name=fs
+    chmod 644 /etc/ceph/ceph.conf
 
-To mount the Ceph file system with ``cephx`` authentication enabled, the
-kernel must authenticate with the cluster. The default ``name=`` option
-is ``guest``.  The mount.ceph helper will automatically attempt to find
-a secret key in a cephx keyring if it's not specified. For example, to
-mount the filesystem as the cephx user ``fs``::
+#. Create a CephX user and get its secret key::
 
-        # mount -t ceph :/ /mnt/mycephfs -o name=fs
+    ssh {user}@{mon-host} "sudo ceph fs authorize cephfs client.foo / rw" | sudo tee /etc/ceph/ceph.client.foo.keyring
 
-The secret can also be specified manually with the ``secret=`` option.::
+   In above command, replace ``cephfs`` with the name of your CephFS, ``foo``
+   by the name you want for CephX user and ``/`` by the path within your
+   CephFS for which you want to allow access to the client and ``rw`` stands
+   for, both, read and write permissions. Alternatively, you may copy the Ceph
+   keyring from the MON host to client host at ``/etc/ceph`` but creating a
+   keyring specific to the client host is better.
 
-	# mount -t ceph :/ /mnt/mycephfs -o name=fs,secret=AQATSKdNGBnwLhAAnNDKnH65FmVKpXZJVasUeQ==
+.. note:: In case you are deploying Ceph in a non-production scenario you
+   may also use admin's secret key instead.
 
-For legacy usage, mount.ceph can be told to read a lone secret from a
-file. For example::
+.. note:: If you get 2 prompts for password while running above any of 2 above
+   command, run ``sudo ls`` (or any other trivial command with sudo)
+   immediately before these commands.
 
-	# mount -t ceph :/ /mnt/mycephfs -o name=fs,secretfile=/etc/ceph/admin.secret
+#. Ensure that the keyring has appropriate permissions::
 
-See `User Management`_ for details on cephx.
+    chmod 600 /etc/ceph/foo.keyring
 
-If you have more than one file system, specify which one to mount using
-the ``mds_namespace`` option, e.g.::
+#. It is recommeneded to have ``mount.ceph`` helper installed. To verify that
+   it is installed, do::
 
-        # mount -t ceph :/ -o name=fs,mds_namespace=myfs
+    ls /sbin/mount.ceph
```

Comment:
`stat ...` is usually clearer.

Reply:
We should also mention that the helper is installed with ceph packages. If installing those packages is not feasible, you can still run the mount command but some arguments must be explicitly provided, like the mon addresses.